parsers/jscript.c

```diff
@@ -2398,10 +2431,42 @@ static bool parsePrototype (tokenInfo *const name, tokenInfo *const token, state
 					skipArgumentList(method_body_token, false,
 									 vStringLength (signature) == 0 ? signature : NULL);
 				else
+				{
+					if ( isType (method_body_token, TOKEN_EQUAL_SIGN) &&
+						! isType (identifier_token, TOKEN_UNDEFINED))
+					{
+						if (prototype_tokens == NULL)
+							prototype_tokens = ptrArrayNew (NULL);
+
+						tokenInfo *const save_token = newToken ();
```

Comment:
I think you can move `strstr(vStringValue (identifier_token->string), ".prototype.")` here.

```C
if (is_prototype(identifier_token))
{
   tokenInfo *const save_token = newToken ();
   copyToken (save_token, identifier_token, true);
   ptrArrayAdd (prototype_tokens, save_token);
}
identifier_token->type = TOKEN_UNDEFINED;
```

So you can reduce the number of token allocations.